server/shared/publishing/helpers.js

```diff
@@ -183,6 +195,18 @@ async function fetchCustomContainers(parent, config) {
   }, []);
 }
 
+async function unpublishDeletedContainers(parent, containers) {
+  const baseUrl = getBaseUrl(parent.repositoryId, parent.id);
+
+  const filePaths = getContainerFilePaths(baseUrl, containers);
+  const publishedFilePaths = await storage.listFiles(baseUrl);
+
+  const redundantFilePaths = difference(publishedFilePaths, filePaths);
+  if (redundantFilePaths.length) await storage.deleteFiles(redundantFilePaths);
+
+  return containers;
```

Comment:
I think you would benefit in readability from the promise chain here rather than async/await with blank spaces.

Reply:
Also, the signature of this method is a bit misleading for me since it's named `unpublishDeletedContainers` while it receives published containers that are to be preserved and is returning that original set of containers as a result 🤔

Reply:
> I think you would benefit in readability from the promise chain here rather than async/await with blank spaces.

Did you mean something like this:
```js
function unpublishDeletedContainers(parent, containers) {
  const baseUrl = getBaseUrl(parent.repositoryId, parent.id);
  const filePaths = getContainersFilePaths(baseUrl, containers);
  return storage
    .listFiles(baseUrl)
    .then(publishedFilePaths => {
      const redundantFilePaths = difference(publishedFilePaths, filePaths);
      if (redundantFilePaths.length) return storage.deleteFiles(redundantFilePaths);
    });
}
```

Reply:
@kronicker ☝️ 👀

Reply:
@abasic 👍